File: python/70.climbing-stairs.py
#
# @lc app=leetcode id=70 lang=python3
#
# [70] Climbing Stairs
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 0:
            return 0
        elif n == 1:
            return 1
        elif n == 2:
            return 2
        dp = [0] * (n + 1)

        for i in range(1, n+1):
            if i == 1:
                dp[i] = 1
            elif i == 2:
                dp[i] = 2
            else: 
                dp[i] = dp[i-2] + dp[i-1]

        return dp[n]
# @lc code=end

if __name__ == "__main__":
    solution = Solution()
    print(solution.climbStairs(1))  # Output: 1
    print(solution.climbStairs(2))  # Output: 2
    print(solution.climbStairs(3))  # Output: 3
    print(solution.climbStairs(4))  # Output: 5
    print(solution.climbStairs(5))  # Output: 8

File: rust/src/70.climbing-stairs.rs
/*
 * @lc app=leetcode id=70 lang=rust
 *
 * [70] Climbing Stairs
 */

use crate::Solution;

// @lc code=start
#[warn(dead_code)]
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        if n == 1{
            1 as i32
        }else if n == 2{
            2 as i32
        }else{
            let mut dp = vec![0; n as usize +1];
            dp[1 as usize] = 1;
            dp[2 as usize] = 2;
            for i in 3..n+1{
                dp[i as usize] = dp[(i-1) as usize] + dp[(i-2) as usize]
            }
            dp[n as usize] as i32
        }
    }
}
// @lc code=end

